frontend/packages/helm-plugin/src/catalog/utils/catalog-utils.tsx

```diff
@@ -107,7 +98,7 @@ export const normalizeHelmCharts = (
         ];
 
         const helmChart = {
-          uid: `${chartRepoName}--${digest}`,
+          uid: `${chartURL}`,
```

Comment:
This might clash again since the same chart can come from multiple chart repositories.

Reply:
How about if we go with `name + randomString` ?

Reply:
This should be good enough.
```suggestion
          uid: `${chartRepoName}--${chartURL}`,
```